native/common.h

```diff
@@ -0,0 +1,81 @@
+#ifndef CV_RS_COMMON_H
+#define CV_RS_COMMON_H
+typedef struct {
+    int32_t x;
+    int32_t y;
+} Point2i;
+
+typedef struct {
+    float x;
+    float y;
+} Point2f;
+
+typedef struct {
+    int width;
+    int height;
+} Size2i;
+
+typedef struct {
+    float width;
+    float height;
+} Size2f;
+
+typedef struct {
+    int32_t x;
+    int32_t y;
+    int32_t width;
+    int32_t height;
+} Rect;
+
+typedef struct {
+    Point2f center;
+    Size2f size;
+    float angle;
+} RotatedRect;
+
+typedef struct {
+    int32_t v0;
+    int32_t v1;
+    int32_t v2;
+    int32_t v3;
+} Scalar;
+
+typedef struct {
+    bool status;
+    uint8_t* buf;
+    size_t size;
+} ImencodeResult;
+
+// Caller is responsible for disposing `error` field
+template<typename T>
+struct Result
+{
+    T value;
+    const char* error;
+
+    static Result<T> FromFunction(std::function<T()> function)
+    {
+        T value;
+        char* error = nullptr;
+        try
+        {
+            value = function();
+        }
+        catch( cv::Exception& e )
+        {
+            const char* err_msg = e.what();
+            auto len = std::strlen(err_msg);
+            error = new char[len + 1];
+            std::strcpy(error, err_msg);
+        }
+        return Result<T>{value, error};
+    }
+};
+
+template<typename T>
+struct CVec
```

Comment:
`CVec {`?

I am mostly following the default LLVM C/C++ styles, see my `.clang-format` as below:

```
BasedOnStyle: llvm
IndentWidth: 4
AllowShortFunctionsOnASingleLine: None
DerivePointerAlignment: false
PointerAlignment: Left
KeepEmptyLinesAtTheStartOfBlocks: false
SpaceAfterCStyleCast: true
SpacesBeforeTrailingComments: 2
AlignAfterOpenBracket: Align
IndentCaseLabels: false
```

Reply:
OK!